### PR TITLE
Add api and rework code for it

### PR DESCRIPTION
**Pull request by it-cio**

### CODE REVIEW

File: src/api/schemas/consumables.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    @validator('compatibility')
+    def compatibility_match(cls, compatibility):
+        return compatibility.lower()
+
+
+class Consumable(ConsumableBase):
+    id: int
+
+    class Config:
+        orm_mode = True
```
Comment (Lehersun) [inline src/api/schemas/consumables.py]: Why do you use so much classes? They complitely do nothing

File: src/api/schemas/equipments.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    @validator('name')
+    def name_match(cls, name):
+        if not re.match(r'^[\w\s.-]+$', name):
+            raise ValueError(f"Name '{name}' is incorrect")
+        return name.lower()
+
+
+class EquipmentCreate(EquipmentBase):
+    pass
```
Comment (Lehersun) [inline src/api/schemas/equipments.py]: It's not clear for me, why you've made empty class?

File: src/api/schemas/users.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    id: int
+    is_admin: bool = False
+
+    class Config:
+        orm_mode = True
+
+
+class UserLogin(BaseModel):
+    email: str
+    password: str
```
Comment (Lehersun) [inline src/api/schemas/users.py]: You need to split different classes to different files or make one class if they have one objective

File: src/api/security.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+auth_handler = AuthHandler()
+
+
+@route.post('/register', status_code=201)
+async def register_new_user(user: UserCreate, db: Session = Depends(get_db)):
+    if db.query(UserModel).filter(UserModel.name == user.name).first():
+        raise HTTPException(status_code=400, detail=f'Username {user.name} is taken')
+    hashed_password = auth_handler.get_password_hash(user.password)
+    db.add(UserModel(name=user.name, email=user.email, password=hashed_password))
+    return {'msg': f'User {user.name} created'}
```
Comment (Lehersun) [inline src/api/security.py]: It's bad idea to do many different operations in one place. You are checking a name and writing data to DB. Also don't see any password validations

File: tests/test_users.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+@temp_database
+def test_delete_user():
+    response = client.delete("/users")
+    assert response.status_code == 200
+    assert response.json() == []
+
+
+def test_existing_temp_database():
+    drop_temp_database()
```
Comment (Lehersun) [inline tests/test_users.py]: I see only API tests of routes and still have the same questions: What about testing the logic? Where are the tests of all functions and classes?

File: src/api/security.py (hunk 4 above)
Comment (it-cio) [reply src/api/security.py]: Created a new module "service" and separated the operations in it. As for password validation, it is written in the user schema.

File: tests/test_users.py (hunk 5 above)
Comment (it-cio) [reply tests/test_users.py]: Add additional tests model classes

File: src/api/schemas/consumables.py (hunk 1 above)
Comment (it-cio) [reply src/api/schemas/consumables.py]: This is schema classes classes, they need to make right api-request. But i make it some simply.

File: src/api/schemas/users.py (hunk 3 above)
Comment (it-cio) [reply src/api/schemas/users.py]: Made this.

File: src/api/schemas/equipments.py (hunk 2 above)
Comment (it-cio) [reply src/api/schemas/equipments.py]: Removed all empty classes.

**Review by Lehersun — APPROVED** [review]

**Pull request merged**